### PR TITLE
Improve supertraits documentation

### PR DESCRIPTION
**Pull request by Havvy**
Some of it feels like it was from the guide. Should all be referency now. And also more cross linking.

### CODE REVIEW

File: src/items/traits.md [hunk 1]
```diff
@@ -10,7 +10,7 @@ interface consists of [associated items], which come in three varieties:
 All traits define an implicit type parameter `Self` that refers to "the type
 that is implementing this interface". Traits may also contain additional type
 parameters. These type parameters (including `Self`) may be constrained by
-other traits and so forth as usual.
+other traits and so forth [as usual].
```
Comment (alercah) [inline src/items/traits.md]: Purely a style nit so feel free to disagree: would it make sense to have [as usual][generics] so it's more clear to someone reading the source where pothole links are going?

**Comment by alercah** [comment]
The number of examples still feels a bit guide-y to me? Perhaps consider collapsing them down into one or two examples.

**Comment by Havvy** [comment]
I disagree. The examples each showcase a different facet of supertraits, not teach how they work. I did end up removing an example of a struct implementing both traits since it didn't showcase anything specific.

File: src/items/traits.md (hunk 1 above)
Comment (Havvy) [reply src/items/traits.md]: I'm heading to work now, so I'll change it when I get home.

**Comment by Havvy** [comment]
Made the "as usual" link easier to follow in markdown source, added a definition for subtract, and pared down the writing for one example.

**Pull request merged**